File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
answer = 0
def isPrime(x):
    global answer
    if x <= 1:
        return
    if x == 2:
        answer += 1
        return
    if x % 2 == 0:
        return
    for i in range(3, int(x**0.5) + 1, 2):
        if x % i == 0:
            return
    answer += 1

def solution(numbers):
    candidate = set()
    for i in range(1, len(numbers) + 1):
        for j in permutations(numbers, i):
            candidate.add(int(''.join(j)))
    
    for i in candidate:
        isPrime(i)        
    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        for (int i = 3; i <= n; i++) {
            if (isPrime(i)) 
                answer += 1;
        }
        
        return answer;
    }
    
    private boolean isPrime(int num) {
    	for (int i = 2; i <= (int) Math.sqrt(num); i++) {
        	if (num % i == 0) {
                return false;
            }
        }
        
        return true;
    }
}